compaction/compaction_manager.cc

```diff
@@ -632,7 +632,9 @@ inline bool compaction_manager::can_proceed(const lw_shared_ptr<task>& task) {
 inline future<> compaction_manager::put_task_to_sleep(lw_shared_ptr<task>& task) {
     cmlog.info("compaction task handler sleeping for {} seconds",
         std::chrono::duration_cast<std::chrono::seconds>(task->compaction_retry.sleep_time()).count());
-    return task->compaction_retry.retry();
+    return task->compaction_retry.retry(task->compaction_data.abort).handle_exception_type([task] (sleep_aborted&) {
+        task->throw_compaction_stopped_exception();
+    });
```

Comment:
yes, it will be indeed better to have make_compaction_stopped_exception() instead which will return exception future.